File: clang/test/CIR/CodeGen/OpenMP/taskyield.cpp
// RUN: %clang_cc1 -std=c++20 -triple x86_64-unknown-linux-gnu -fopenmp-enable-irbuilder -fopenmp -fclangir -emit-cir %s -o %t.cir
// RUN: FileCheck --input-file=%t.cir %s

// CHECK: cir.func
void omp_taskyield_1(){
// CHECK: omp.taskyield
  #pragma omp taskyield
}

File: flang/test/Lower/OpenMP/taskyield.f90
!RUN: %flang_fc1 -emit-hlfir -fopenmp %s -o - | FileCheck %s

!CHECK-LABEL: @_QPomp_taskyield
subroutine omp_taskyield
  !CHECK: omp.taskyield
  !$omp taskyield
  !CHECK: fir.call @_QPfoo() {{.*}}: () -> ()
  call foo()
  !CHECK: omp.taskyield
  !$omp taskyield
end subroutine omp_taskyield
